akka-docs/src/main/paradox/scala/stream/stream-error.md

```diff
@@ -101,4 +101,39 @@ Java
 :   @@snip [IntegrationDocTest.java]($code$/java/jdocs/stream/IntegrationDocTest.java) { #email-addresses-mapAsync-supervision }
 
 If we would not use `Resume` the default stopping strategy would complete the stream
-with failure on the first @scala[`Future`] @java[`CompletionStage`] that was completed @scala[with `Failure`]@java[exceptionally].
\ No newline at end of file
+with failure on the first @scala[`Future`] @java[`CompletionStage`] that was completed @scala[with `Failure`]@java[exceptionally].
+
+## Delayed restarts with a backoff stage
+
+Just as Akka provides the @ref:[backoff supervision pattern for actors](../general/supervision.md#backoff-supervisor), Akka streams
+also provides a `RestartSource`, `RestartSink` and `RestartFlow` for implementing the so-called *exponential backoff 
+supervision strategy*, starting a stage again when it fails, each time with a growing time delay between restarts.
+
+This pattern is useful when the stage fails <a id="^1" href="#1">[1]</a> because some external resource is not available,
+and we need to give it some time to start-up again. One of the prime examples when this is useful is
+when a WebSocket connection fails due to the HTTP server it's running on going down, perhaps because it is overloaded. 
+By using an exponential backoff, we avoid going into a tight reconnect look, which both gives the HTTP server some time
+to recover, and it avoids using needless resources on the client side.
+
+> <a id="1" href="#^1">[1]</a> A failure can be indicated in two different ways; by the stream completing or failing.
+
+The following snippet shows how to create a backoff supervisor using `akka.stream.scaladsl.RestartSource`
+which will supervise the given `Source`. The `Source` in this case is a stream of Server Sent Events, 
+produced by akka-http. If the stream fails at any point, the request will be made again, in increasing 
+intervals of 3, 6, 12, 24 and finally 30 seconds:
```

Comment:
`finally 30 seconds (at which it will remain capped due to the maxBackoff parameter)`